numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1078,6 +1079,18 @@ gentype_richcompare(PyObject *self, PyObject *other, int cmp_op)
         }
     }
 
+    // Check if we need to defer to the other object's method xref GH#17017
+    if (PyObject_HasAttrString(other, "__array_priority__") &&
+            PyObject_HasAttrString(self, "__array_priority__")) {
+
+        self_prio = PyArray_GetPriority(self, NPY_SCALAR_PRIORITY);
+        other_prio = PyArray_GetPriority(other, NPY_SCALAR_PRIORITY);
+        if (other_prio > self_prio) {
+            Py_INCREF(Py_NotImplemented);
+            return Py_NotImplemented;
+        }
+    }
+
```

Comment:
```suggestion
   RICHCMP_GIVE_UP_IF_NEEDED(self, other);

```
Thanks for looking into this!  I think we should be able to just use this macro.  Now that I look at it, I am surprised it is not already used.

Reply:
definitely nicer to not re-implement.  Would my way have worked?  I was pretty proud of figuring out where/how to do this.

Reply:
Yeah, this is mostly right.  `RICHCMP_GIVE_UP_IF_NEEDED` does a little bit more, because it also/first checks `__array_ufunc__`, and it has a bit of short-cutting for builtin Python types/fast-paths, although I am not sure how important that part is.

I guess the `HasAttrString` was unnecessary, since `PyArray_GetPriority` should do that right (although IIRC it swallows a bit many errors I think, but not much too many).

The macro also makes sure we do not defer if we are a superclass, because in that case the other object already had a chance, but I am not sure that matters here.